velox/vector/DecodedVector.cpp

```diff
@@ -441,16 +441,17 @@ void DecodedVector::setBaseDataForBias(
 }
 
 namespace {
-bool isWrapper(VectorEncoding::Simple encoding) {
-  return encoding == VectorEncoding::Simple::SEQUENCE ||
-      encoding == VectorEncoding::Simple::DICTIONARY;
+bool isWrapper(const BaseVector& vector) {
+  return vector.encoding() == VectorEncoding::Simple::SEQUENCE ||
+      vector.encoding() == VectorEncoding::Simple::DICTIONARY ||
+      (vector.encoding() == VectorEncoding::Simple::CONSTANT &&
+       vector.valueVector() != nullptr);
 }
-
 /// Returns true if 'wrapper' is a dictionary vector wrapping non-dictionary
 /// vector.
 bool isOneLevelDictionary(const BaseVector& wrapper) {
   return wrapper.encoding() == VectorEncoding::Simple::DICTIONARY &&
-      !isWrapper(wrapper.valueVector()->encoding());
+      !isWrapper(*wrapper.valueVector());
```

Comment:
The helper function isOneLevelDictionary is used only in DecodedVector::dictionaryWrapping. It is supposed to return true in case when the 'wrapper' vector is a Dictionary-over-Flat and therefore it is ok to re-use the indices buffer from the 'wrapper' vector. Otherwise, we need to make a new indices buffer.

Therefore, I suggest to remove isWrapper helper function (it is only used by isOneLevelDictionary) and change the logic of isOneLevelDictionary to something like this:

```
return wrapper.encoding() == VectorEncoding::Simple::DICTIONARY &&
      wrapper.valueVector()->isFlatEncoding();
```

CC: @kagamiori @bikramSingh91

Reply:
@mbasmanova Thanks for elaborating the context. Your suggested change sounds reasonable to me.
I echo your comment - furthermore, I prefer to rename the help function from `isOneLevelDictionary()` to `isDictionaryOverFlat()` because the case "dict over constant(scalar type)" is literally a "one level dictionary" as well, which creates some confusion.